### ISSUE TITLE
Build failing with updateBuildableProjectDepsInPackageJson is not found in schema

### ISSUE THREAD

**Issue by vicatcu** [nrwl/nx#27325]
### Current Behavior

I am pretty sure I built my app successfully in the last month or two, but something must have changed under the covers (maybe the VSCode plugin? I don't know) and now I'm getting this error on all the libraries in my Nx monorepo when I try and build my app. So my two questions are, (1) what could have changed to break my ability to build my project, and (2) how do I repair whatever is wrong now?

### Expected Behavior

`npx nx run my-app:build --configuration=development` should succeed without errors.

### Steps to Reproduce

1. npx nx run my-app:build --configuration=development
2. See updateBuildableProjectDepsInPackageJson errors in output

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 18.20.2
   OS     : linux-x64
   npm    : 10.5.0
   
   nx                 : 17.0.0
   @nx/js             : 17.0.0
   @nx/jest           : 17.0.0
   @nx/linter         : 17.0.0
   @nx/eslint         : 17.0.0
   @nx/workspace      : 17.0.0
   @nx/angular        : 17.0.0
   @nx/cypress        : 17.0.0
   @nx/devkit         : 17.0.0
   @nx/eslint-plugin  : 17.0.0
   @nrwl/tao          : 17.0.0
   @nx/web            : 17.0.0
   @nx/webpack        : 17.0.0
   typescript         : 5.1.6
   ---------------------------------------
   Community plugins:
   @ionic/angular        : 7.1.3
   @nxtend/capacitor     : 13.0.0
   @nxtend/ionic-angular : 13.1.0
```

### Failure Logs

```shell
> nx run utility:build:development
 >  NX   'updateBuildableProjectDepsInPackageJson' is not found in schema
   ✖    2/2 dependent project tasks failed (see below)
```

### Package Manager Version

npm 10.5.0

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

VSCode 1.91.1
Nx Console VSCode Plugin 18.25.0

**Comment by vicatcu** [comment]
I got the project to build again by searching the entire project for updateBuildableProjectDepsInPackageJson and removing all occurrences of:

```
,
        "updateBuildableProjectDepsInPackageJson": true
```

with an empty string... Those were basically in every project.json file I guess it must have been in there because a migration script? It remains unclear to me whether this is the correct solution to the problem, so I hope one of the devs will chime in and set me straight one way or the other.

**Comment by leosvelperez** [comment]
Thanks for reporting this!

You did the right thing by removing the option. That's the expected change.

There's a migration to do so, but due to a mistake, it was only registered to run in Nx v17.0.4 onwards. Unfortunately, we can't amend published versions, so running `nx migrate 17.0.4` (or to a later version) would do it for you.

I'm closing this since you've solved the issue and there's a migration in a later version to apply the change.
